test/test_linalg.py

```diff
@@ -2826,7 +2827,7 @@ def _test_svd_helper(self, shape, some, col_maj, device, dtype):
         #   then the corresponding column of the V has to be changed.
         # Thus here we only compare result[..., :m].abs() from CPU and device.
         for x, y in zip(cpu_result, device_result):
-            self.assertEqual(x[..., :m].abs(), y[..., :m].abs(), atol=1e-5, rtol=0)
+            self.assertEqual(x[..., :m].abs().to(dtype=dtype), y[..., :m].abs().to(dtype=dtype))
```

Comment:
Instead of converting the dtype here I think this can just set `exact_dtype=False` when calling `assertEqual`